qiita_db/metadata_template.py

```diff
@@ -1051,6 +1085,27 @@ def get_filepaths(self, conn_handler=None):
 
         return [(fpid, base_fp(fp)) for fpid, fp in filepath_ids]
 
+    def categories(self):
+        """Get the categories associated with self
+
+        Returns
+        -------
+        set
```

Comment:
One possible use case would be for round-tripping users' metadata files (see issue #790), so they can download exactly what they uploaded at a future date